src/linker/configuration.py

```diff
@@ -36,8 +36,15 @@ def __init__(
     def get_step_directory(self, step_name: str) -> Path:
         # TODO: move this into proper config validator
         implementation = self.pipeline["steps"][step_name]["implementation"]
-        if implementation == "pvs_like_python":
-            # TODO: stop hard-coding filepaths
+        implementation_names = next(
```

Comment:
Is there a reason we don't want to store the implementation names in a dict or a list for easier retrieval?

Reply:
My intent was to avoid having to repeat ourselves, writing the names of the implementations in multiple places. You have to specify implementation in the pipeline.yaml and the directory name (assuming you just map those to directories, which seems intuitive to me). The alternative means another constant.

Reply:
I like not duplicating it in a constant, but maybe Rajan's comment is more about how much code duplication there is here? Why not this?

```python
implementations_path = (
    Path(os.path.realpath(__file__)).parent.parent.parent
    / "steps"
    / step_name
    / "implementations"
)
if implementation in os.walk(implementations_path):
    step_dir = implementations_path / implementation
else:
    raise NotImplementedError(...)
```